packages/auto-label/src/auto-label.ts

```diff
@@ -0,0 +1,256 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { Application, Octokit } from 'probot';
+import { GitHubAPI } from 'probot/lib/github';
+import * as path from 'path';
+import e = require('express');
+import { IssuesListLabelsOnIssueResponseItem } from '@octokit/rest';
+
+const fs = require('fs');
+const colorsData = JSON.parse(
+  fs.readFileSync(path.resolve(__dirname, '../../src/colors.json'), 'utf8')
+);
+
+interface JSONData {
+  num_total_prs: number;
+  num_open_p1s: number;
+  num_open_questions: number;
+  num_open_p0s: number;
+  language: string;
+  api_shortname: string;
+  github_label: string;
+  num_open_p2s: number;
+  is_tracking_issues: boolean;
+  repo: string;
+  is_tracking_samples: boolean;
+  num_open_issues: number;
+  num_total_issues: number;
+  num_open_prs: number;
+  issue_score: number;
+  num_slo_violations: number;
+  num_commits: number;
+}
+
+handler.addLabels = async function addLabels(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  issueNumber: number,
+  labels: string[]
+) {
+  try {
+    const data = await github.issues.addLabels({
+      owner,
+      repo,
+      issue_number: issueNumber,
+      labels,
+    });
+    return data;
+  } catch (err) {
+    return null;
+  }
+};
+
+handler.checkExistingLabels = async function checkExistingLabels(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  name: string
+) {
+  try {
+    const data = await github.issues.getLabel({
+      owner,
+      repo,
+      name,
+    });
+    return data.data.name;
+  } catch (err) {
+    return null;
+  }
+};
+
+handler.createLabel = async function createLabel(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  name: string,
+  color: string
+) {
+  try {
+    const data = await github.issues.createLabel({
+      owner,
+      repo,
+      name,
+      color,
+    });
+    return data;
+  } catch (err) {
+    return null;
+  }
+};
+
+handler.checkExistingIssueLabels = async function checkExistingIssueLabels(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  issueNumber: number
+) {
+  try {
+    const data = await github.issues.listLabelsOnIssue({
+      owner,
+      repo,
+      issue_number: issueNumber,
+    });
+    if (data.data === undefined) {
+      return null;
+    } else {
+      return data.data;
+    }
+  } catch (err) {
+    return null;
+  }
+};
+
+handler.callStorage = async function callStorage(
+  bucketName: string,
+  srcFileName: string,
+  destFileName: string
+) {
+  const { Storage } = require('@google-cloud/storage');
+  const storage = new Storage();
+  const destFilename = path.resolve(destFileName);
+
+  const options = {
+    // The path to which the file should be downloaded, e.g. "./file.txt"
+    destination: destFilename,
```

Comment:
you can actually just download into memory like so:

```js
   // for mono-repos we populate a list of all apis and products,
    // since each repo might include multiple products:
    console.log(`populating all api labels for ${repoPath}`);
    const apis = await storage
      .bucket('devrel-prod-settings')
      .file('apis.json')
      .download();
    return JSON.parse(apis[0].toString()) as GetApiLabelsResponse;
```

:point_up: I had to crack open the reference docs, wasn't super well documented.